am-lib/src/test/java/integration/uk/gov/hmcts/reform/amlib/GrantAccessIntegrationTest.java

```diff
@@ -53,4 +56,18 @@ void whenCreatingDuplicateResourceAccessEntryIsOverwritten() {
 
         assertThat(databaseHelper.countExplicitPermissions(resourceId)).isEqualTo(1);
     }
+
+    @Test
+    void whenCreatingResourceForMultipleUsersShouldAppearInDatabase() {
+        Map<JsonPointer, Set<Permission>> multipleAttributePermissions =
+            ImmutableMap.<JsonPointer, Set<Permission>>builder()
+                .put(JsonPointer.valueOf(""), EXPLICIT_READ_CREATE_UPDATE_PERMISSIONS)
+                .put(JsonPointer.valueOf("/name"), EXPLICIT_READ_CREATE_UPDATE_PERMISSIONS)
+                .build();
+
+        service.grantExplicitResourceAccess(createGrant(resourceId, ACCESSOR_IDS, multipleAttributePermissions));
+
+        assertThat(databaseHelper.findExplicitPermissions(resourceId)).hasSize(4)
+            .extracting(ExplicitAccessRecord::getAccessorId).contains("y", "z");
```

Comment:
Be more string. We wanna make sure only Y and Z are there.

Reply:
👍